### PR TITLE
Fetching Data over FakeJSON

### LINKED ISSUE #85: Fetching Data from URL
MicroByte Title: Fetching Data from URL

Domain: Android App Development

Scope of the learners:
1. Learners will learn how to fetch data from the given URL.
2. Able to learn how to use API calls.

Activities:
1. Allow Permissions in AndroidManifest.xml
2. Adding Dependencies
3. Coding Phase in MainActivity.java

### PR DESCRIPTION
**Pull request by deepak-prajapatii**
# Closes #85 

Checklist:
- [x] Added activities for learners
- [x] Didn't give the solutions to the added activities straight away
- [x] Followed a LearnByDoing approach
- [x] Clean and well-documented code (if source code files are involved)
- [x] Added images for better understanding if deemed necessary

### CODE REVIEW

**Comment by deepak-prajapatii** [comment]
@Ak-Shaw @amoghrajesh Today is the last day of the hacktberfest. So Review my PR ASAP, if there are errors so i can update my pr within the time.

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @deepak-prajapatii!
I will be honest with you. This content is too small for a MicroByte. It will get over in 20-25 mins max. 

1. I require you to add more content to extend it to 1 hr.

2. In activity 1, you say add permissions like: "Usage on Internet". Is this the only one to be added? If more is needed, kindly mention it clearly.

3. Need more description of all the activities. They are very less and vague right now.

4. You can't just add 'Complete all activities' as `Micro Challenge`. Remove that and add something else there.

Make these changes asap. I will re review it after

**Comment by Kevinpaulose05** [comment]
https://github.com/Crio-Bytes/Miscellaneous/pull/89#issuecomment-719889839

**Comment by deepak-prajapatii** [comment]
@amoghrajesh  I updated my PR. Please review it again.

File: Fetching data over FakeJSON/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Introduction
+
+Hi Geeks, I'm sure you may heard about the API's(Applciation Programming Interface) and why we use it? Okay, I give you a short introduction about the API's. So, API  is a computing interface which defines interactions between multiple software intermediaries. In other simple words, It defines the kinds of calls or requests that can be made, how to make them, the data formats that should be used, the conventions to follow, etc.
```
Comment (amoghrajesh) [inline Fetching data over FakeJSON/README.md]: From "Okay, I will give you a short intro..." make it into points.

File: Fetching data over FakeJSON/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# Introduction
+
+Hi Geeks, I'm sure you may heard about the API's(Applciation Programming Interface) and why we use it? Okay, I give you a short introduction about the API's. So, API  is a computing interface which defines interactions between multiple software intermediaries. In other simple words, It defines the kinds of calls or requests that can be made, how to make them, the data formats that should be used, the conventions to follow, etc.
+
+You would also need to know about the JSON and how it works? So basically, JSON stands for JavaScript Object Notation. JSON is a lightweight format for storing and transporting data. JSON is often used when data is sent from a server to a web page.
+
+In this microbyte, we will going to fetch data from the given URL. URL is taken from the website [FAKEJSON Placeholder](https://jsonplaceholder.typicode.com/). This website allows  you to use fake Online REST API for testing and prototyping with no cose. It is absolutely free.
```
Comment (amoghrajesh) [inline Fetching data over FakeJSON/README.md]: Make these into points too

File: Fetching data over FakeJSON/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+3. Use of Volley library and it's predefined methods and object.
+4. Fetching data from the given URL.
+
+
+# Activites
+
+## Activity 1: Allow Permissions and adding dependencies
+
+There are some permissions which would required to fetching the data from the url. You can add permissions in your "AndroidManifest.xml" file. Permissions like Usage of Internet will require to fetching the data.
+Now it's time to use Volley Library, you must need to add the ependencies of the library. You can get the library dependencies from [there](https://developer.android.com/training/volley). Don't worry if you don't know how to add the dependencies in android studio. Go through the reference link and explore.<br/>
```
Comment (amoghrajesh) [inline Fetching data over FakeJSON/README.md]: dependencies* - typo

File: Fetching data over FakeJSON/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+[Reference link 1](https://medium.com/techpin/https-medium-com-s2purno-volley-get-and-post-jsonobject-request-46fb8a46f799)<br/>
+[Reference link 2](https://stackoverflow.com/questions/19837820/volley-jsonobjectrequest-post-request-not-working)<br/>
+[How to use the Logcat Terminal for debugging](https://developer.android.com/studio/debug/am-logcat)
+<br/><br/><br/>
+There are some more screenshots provided to you in the images folder for your convinience.
+
+
+## Activity 3: Display content to the user's screen
+
+We were always used the "Logcat Terminal" to see the response. What if the response is required to show the details to the user? I mean to say that you was successfully fetching the data and see the response in "Logcat Terminal". Now it's time to show the details to the user by creating a beautiful UI/LAYOUT of the app. There are some reference links which will helps you for sure.<br/><br/>
```
Comment (amoghrajesh) [inline Fetching data over FakeJSON/README.md]: We have always used the "Logcat terminal"* - typo

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @deepak-prajapatii!
There is a good improvement compared to the older version. I appreciate your quick response.

There are some smaller changes still:
1. You are telling the user to use "Logcat Terminal" but you have made no mention of what it is or how to power it on and/or use it. Kindly add that.
2. It doesn't look good to use Reference names as "Reference 1" or "Reference 2". Change it to the title it points to. See screenshot below:
image
3. In Activity 2, you have provided a snippet of code. Where is that code present/ Where is it coming from/ What does it do/ How to use it? You need to take care of these questions as it could be confusing to users.

For now, these are the corrections required. 
Make them asap and we shall re-review it asap

**Comment by deepak-prajapatii** [comment]
@amoghrajesh Hey, I updated my PR again according to you. Please review it again. If there is something wrong, guide me.

**Comment by deepak-prajapatii** [comment]
@amoghrajesh  Hey, I updated my PR again according to you. Please review it again ASAP. If there is something wrong, guide me.

**Review by archithdwij — COMMENTED** [review]
Hi @deepak-prajapatii 
You have a good topic in hand but the MicroByte you have created doesn't make justice to the quality. The MB you have created here is missing the following - 

- The way you have created the content is not helping learners gain knowledge about the topic. You have provided the  definitions about API, JSON which is all good. But the content is not wholesome and is missing engaging activites. You have provided the list of points that any learner will gain, from which the first two (API, JSON) is given as a defintion. The following content doesn't contain a process flow. 

- When it comes to the activity part, all of them contains only reference link which learner has to visit and learn. You have not any learning tasks that you provide that engages the learner, thereby failing the approach that every MB needs. You could have explained why specifc library is used, what kind of role does it play in making the process click. 

When you say "You have sucessfully learned how to use APIs and fetch data", you are saying the learner will only know how to use it, but does not understand the backend. Kindly requesting you to reasses the missing blocks, and restructure you content that meets and better, overcomes your previous PR's standard. 

Though the team tried to guide you by requesting changes, you haven't scaled the content up. Refer to the previous comments to gather the idea and that will help you in restructuring in a specific way.

With this, I am closing this PR. You can still redo your work and astonish us with your restructured premium content which we know you can do. 

Thank you on the behalf of our team for participating in the #IBD - Hacktober edition and looking forward for your presence and contributions in the upcoming events.

**Pull request closed without merging**